### PR TITLE
macOS Sonoma (14.0) / Xcode 15.0 — Compatibility Fixes + Docs

### PR DESCRIPTION
**Pull request by coreh**
# Objective

Improve compatibility with macOS Sonoma and Xcode 15.0.

## Solution

- Adds the workaround by @ptxmac to ignore the invalid window sizes provided by `winit` on macOS 14.0
  - This still provides a slightly wrong content size when resizing (it fails to account for the window title bar, so some content gets clipped at the bottom) but it's _much better_ than crashing.
- Adds docs on how to work around the `bindgen` bug on Xcode 15.0.

## Related Issues:

- https://github.com/RustAudio/coreaudio-sys/issues/85
- https://github.com/rust-windowing/winit/issues/2876

---

## Changelog

- Added a workaround for a `winit`-related crash under macOS Sonoma (14.0)

### CODE REVIEW

**Comment by Vrixyz** [comment]
Is there a follow up issue to track when this « hack » + « slightly wrong » resizing can be fixed ?

Maybe next version of winit will help ? In the meantime that PR sounds great 👍

**Comment by coreh** [comment]
The fix for it is already merged to `winit`. However, there hasn't been a new release with it yet.

It _should_ will be included in 0.29, so perhaps we could just use your PR (#8745) to track that? (By removing the workaround on your branch)

**Comment by Vrixyz** [comment]
I’d rather avoid my PR to be a tracking issue for this fix, because:
- #8745 might not be accepted
- #8745 will probably not get the fix in (the PR is already big, and that’s additional testing…)

I added a link though, so we’re not forgetting it, I’ll create a standalone issue or PR after we’re done with #8745 if need be.

**Comment by rparrett** [comment]
> The fix for it is already merged to winit. However, there hasn't been a new release with it yet.

The fix seems relatively simple -- I wonder if they would be willing to backport it for a minor 0.28 release.

File: docs/macos_dependencies.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  --- stderr
+  thread 'main' panicked at '"enum_(unnamed_at_/Applications/Xcode_app/Contents/Developer/Platforms/MacOSX_platform/Developer/SDKs/MacOSX14_0_sdk/usr/include/MacTypes_h_382_1)" is not a valid Ident', <path to your user folder>/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.56/src/fallback.rs:811:9
+  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+```
+
+There's an upstream [issue](https://github.com/RustAudio/coreaudio-sys/issues/85) in `coreaudio-sys` to update `bindgen`. In the mean time you can work around this by installing clang from [Homebrew](https://brew.sh):
+
+```sh
+brew install llvm@15
+```
```
Comment (simlay) [inline docs/macos_dependencies.md:31]: Sorry for being late to see the coreaudio-sys issue. A new minor version of coreaudio-sys has been published with an updated version of bindgen so this work around shouldn't be necessary.

Comment (coreh) [reply docs/macos_dependencies.md:31]: No problem, that's even better!

Thanks for updating it, I'll update the PR bumping the version.

Comment (coreh) [reply docs/macos_dependencies.md]: Actually scratch that, I forgot that Cargo.lock is not tracked for libraries, so there's no need to update anything 🥳 

I just removed the special instructions

**Review by mockersf — APPROVED** [review]

**Review by zaycev — APPROVED** [review]

**Review by cart — APPROVED** [review]

**Pull request merged**